src/app/pre-sale/providers/presale-provider.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useMemo, useState } from 'react'
+
+import { preSaleTokens } from '@/app/pre-sale/constants'
+import { MetaverseIndex, Token, WSTETH } from '@/constants/tokens'
+import { formatAmount } from '@/lib/utils'
+import { IndexApi } from '@/lib/utils/api/index-api'
+
+interface PresaleData {
+  currencyToken: Token
+  data: {
+    tvl: number
+  }
+  formatted: {
+    prtRewards: string
+    indexRewards: string
+    tvl: string
+  }
+}
+
+export function usePresaleData(symbol: string): PresaleData {
+  const presaleToken = preSaleTokens.find((token) => token.symbol === symbol)
+  const currencyToken = WSTETH
+
+  const [tvl, setTvl] = useState(0)
+
+  const tvlFormatted = useMemo(
+    () => `${formatAmount(tvl)} ${currencyToken.symbol}`,
+    [currencyToken, tvl],
+  )
+
+  useEffect(() => {
+    // FIXME: use hyETH (presale token)
+    const token = MetaverseIndex
+    const fetchTvl = async () => {
+      try {
+        const indexApi = new IndexApi()
+        const res = await indexApi.get(`/${token.symbol}/marketcap`)
+        console.log('fetch:', res)
+        setTvl(res.marketcap)
+      } catch (err) {
+        console.log('Error fetching tvl', err)
+      }
+    }
+    fetchTvl()
+  }, [symbol])
+
+  return {
+    currencyToken,
+    data: {
+      ...presaleToken,
+      tvl,
+    },
+    formatted: {
+      prtRewards: '',
+      indexRewards: '',
+      tvl: tvlFormatted,
+    },
+  }
+}
```

Comment:
As mentioned on the previous PR, the idea of this provider would be to return all related data (raw + formatted). This can be used in the presale cards and widget. The widget would mainly use the TVL.